core/ui/src/lib/components/repo/ModifiedFilesCard.svelte

```diff
@@ -208,11 +208,15 @@
 				(file) => !changeSet.files.some((csFile) => csFile.path === file.path)
 			);
 		} else {
-			// avoid duplicates
+			// avoid duplicates and files that don't match the current search input
 			selectedFiles = [
 				...selectedFiles,
 				...changeSet.files.filter(
-					(file) => !selectedFiles.some((selectedFile) => selectedFile.path === file.path)
+					(file) =>
+						!selectedFiles.some((selectedFile) => selectedFile.path === file.path) &&
+						(searchInput.length < 3 ||
+							file.path.toLowerCase().includes(searchInput.toLowerCase()) ||
+							file.displayName.toLowerCase().includes(searchInput.toLowerCase()))
```

Comment:
before this change, select all would select everything even if you had filtered files via the search bar.